chainstate/test-suite/src/tests/chainstate_accounting_storage_tests.rs

```diff
@@ -0,0 +1,579 @@
+    outpoint_id: OutPointSourceId,
+    amount_to_stake: Amount,
+    pub_key: &PublicKey,
+) -> (SignedTransaction, PoolId) {
+    let (_, vrf_pub_key) = VRFPrivateKey::new(VRFKeyKind::Schnorrkel);
+    let tx_output = TxOutput::new(
+        OutputValue::Coin(amount_to_stake),
+        OutputPurpose::StakePool(Box::new(StakePoolData::new(
+            anyonecanspend_address(),
+            None,
+            vrf_pub_key,
+            pub_key.clone(),
+            0,
+            Amount::ZERO,
+        ))),
+    );
+
+    let input0_outpoint = OutPoint::new(outpoint_id, 0);
+    let pool_id = pos_accounting::make_pool_id(&input0_outpoint);
+    let tx = TransactionBuilder::new()
+        .add_input(
+            TxInput::new(input0_outpoint.tx_id(), input0_outpoint.output_index()),
+            empty_witness(rng),
+        )
+        .add_output(tx_output)
+        .build();
+    (tx, pool_id)
+}
+
+// Process a tx with a stake pool. Check that new pool balance and data are stored
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn store_pool_data_and_balance(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let storage = Store::new_empty().unwrap();
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).with_storage(storage.clone()).build();
+        let amount_to_stake = Amount::from_atoms(100);
+        let (_, pub_key) = PrivateKey::new_from_rng(&mut rng, KeyKind::RistrettoSchnorr);
+
+        let (tx, pool_id) =
+            make_tx_with_stake_pool_from_genesis(&mut rng, &mut tf, amount_to_stake, &pub_key);
+        let tx_utxo_outpoint =
+            OutPoint::new(OutPointSourceId::Transaction(tx.transaction().get_id()), 0);
+
+        let block = tf.make_block_builder().add_transaction(tx).build();
+        let block_id = block.get_id();
+        tf.process_block(block, BlockSource::Local).unwrap();
+
+        // check that result is stored
+        let db_tx = storage.transaction_ro().unwrap();
+
+        // utxo is stored
+        db_tx.get_utxo(&tx_utxo_outpoint).expect("ok").expect("some");
+        assert_eq!(
+            db_tx.get_undo_data(block_id).expect("ok").expect("some").tx_undos().len(),
+            1
+        );
+
+        let expected_tip_storage_data = pos_accounting::PoSAccountingData {
+            pool_data: BTreeMap::from([(pool_id, PoolData::new(pub_key, amount_to_stake))]),
+            pool_balances: BTreeMap::from([(pool_id, amount_to_stake)]),
+            delegation_balances: Default::default(),
+            delegation_data: Default::default(),
+            pool_delegation_shares: Default::default(),
+        };
+
+        assert_eq!(
+            storage.read_accounting_data_tip().unwrap(),
+            expected_tip_storage_data
+        );
+
+        assert!(storage.read_accounting_data_sealed().unwrap().is_empty());
+    });
+}
+
+// Create block1 from genesis and block2 from block1 using default chain config.
```

Comment:
We can't guarantee that the default chain config won't change by the release. Maybe we should make this specific?